project/ws/viewer/src/lib/plugins/html/html.component.ts

```diff
@@ -358,26 +358,30 @@ export class HtmlComponent implements OnInit, OnChanges, OnDestroy {
-            // tslint:disable-next-line:max-line-length
-            `${this.generateUrl(this.htmlContent.streamingUrl)}/${this.htmlContent.initFile}?timestamp='${new Date().getTime()}`
-          )
+        if (this.htmlContent.streamingUrl.includes(environment.azureHost)) {
+          this.iframeUrl = this.domSanitizer.bypassSecurityTrustResourceUrl(this.htmlContent.streamingUrl)
         } else {
-          if (environment.production) {
+          if (this.htmlContent.streamingUrl && this.htmlContent.initFile) {
             this.iframeUrl = this.domSanitizer.bypassSecurityTrustResourceUrl(
-              // tslint:disable-next-line: max-line-length
-              // `${environment.azureHost}/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
-              // tslint:disable-next-line: max-line-length
-              `${environment.azureHost}/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
+              // tslint:disable-next-line:max-line-length
+              `${this.generateUrl(this.htmlContent.streamingUrl)}/${this.htmlContent.initFile}?timestamp='${new Date().getTime()}`
             )
           } else {
-            this.iframeUrl = this.domSanitizer.bypassSecurityTrustResourceUrl(
-              // tslint:disable-next-line: max-line-length
-              // `${environment.azureHost}/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
-              // tslint:disable-next-line: max-line-length
-              `/abcd/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
-            )
+            if (environment.production) {
+              this.iframeUrl = this.domSanitizer.bypassSecurityTrustResourceUrl(
+                // tslint:disable-next-line: max-line-length
+                // `${environment.azureHost}/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
+                // tslint:disable-next-line: max-line-length
+                `${environment.azureHost}/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
+              )
+            } else {
+              this.iframeUrl = this.domSanitizer.bypassSecurityTrustResourceUrl(
+                // tslint:disable-next-line: max-line-length
+                // `${environment.azureHost}/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
+                // tslint:disable-next-line: max-line-length
+                `/abcd/${environment.azureBucket}/content/html/${this.htmlContent.identifier}-snapshot/index.html?timestamp='${new Date().getTime()}`
```

Comment:
abcd , we need to add?

Reply:
this is used for local proxy it is needed